src/test/ui/wf/issue-87495.rs

```diff
@@ -0,0 +1,11 @@
+// Regression test for the ICE described in #87495.
+
+#![allow(non_camel_case_types,bare_trait_objects)]
+
+trait u32 {
```

Comment:
Could you rename this to something other than `u32,` to avoid confusion? (I've verified that doing so still produces the same ICE).